### PR TITLE
sys-process/lsof: fix build issues

### PR DESCRIPTION
**Pull request by AngryLoki**
This PR fixes compilation issue fir HIP-enabled clang and [pg0303](https://projects.gentoo.org/qa/policy-guide/installed-files.html#pg0303) QA warning about redundant lo file.

Upstream bug: https://github.com/lsof-org/lsof/issues/305

Closes: https://bugs.gentoo.org/919253
Closes: https://bugs.gentoo.org/920370

### CODE REVIEW

File: sys-process/lsof/files/lsof-4.98.0-fix-clang-version-parser.patch [hunk 1]
```diff
@@ -0,0 +1,15 @@
+https://bugs.gentoo.org/919253
```
Comment (thesamesam) [inline sys-process/lsof/files/lsof-4.98.0-fix-clang-version-parser.patch:1]: Can you send this upstream as a PR please?

File: sys-process/lsof/lsof-4.98.0.ebuild [hunk 2]
```diff
@@ -28,6 +28,15 @@ BDEPEND="
...
 RESTRICT="test"
 
+PATCHES=(
+	"${FILESDIR}"/${PN}-4.98.0-fix-clang-version-parser.patch
+)
+
+src_prepare() {
+	default
+	eautoreconf
+}
```
Comment (thesamesam) [inline sys-process/lsof/lsof-4.98.0.ebuild:38]: Add a comment above the eautoreconf noting that it's just there for the patch, so we know to drop it later.

Comment (AngryLoki) [reply sys-process/lsof/lsof-4.98.0.ebuild:40]: Done

File: sys-process/lsof/files/lsof-4.98.0-fix-clang-version-parser.patch (hunk 1 above)
Comment (AngryLoki) [reply sys-process/lsof/files/lsof-4.98.0-fix-clang-version-parser.patch:1]: Done and already merged in https://github.com/lsof-org/lsof/pull/306

**Comment by thesamesam** [comment]
Thanks!

**Pull request closed without merging**